src/interface/histogram.jl

```diff
@@ -90,12 +90,13 @@ function histogram(
         kw...)
     edges, counts = hist.edges[1], hist.weights
     labels = Vector{String}(undef, length(counts))
-    binwidth = typeof(edges.step) == Float64 ? edges.step : edges.step.hi
+    binwidths = diff(edges)
```

Comment:
Since we don't currently have CI or test backed, so just a check, is `length(diff(edges)) == length(counts)`?

Reply:
This should always be true since the `Histogram` object itself requires this relationship:
```julia
julia> StatsBase.Histogram([1,2,3],[1,2,3])
ERROR: Histogram edge vectors must be 1 longer than corresponding weight dimensions
```